File: app/src/main/java/com/stevenfrew/beatprompter/event/MIDIEvent.java
package com.stevenfrew.beatprompter.event;

import com.stevenfrew.beatprompter.midi.OutgoingMessage;
import com.stevenfrew.beatprompter.midi.EventOffset;

import java.util.ArrayList;
import java.util.List;

public class MIDIEvent extends BaseEvent {
    public List<OutgoingMessage> mMessages;
    public EventOffset mOffset;

    public MIDIEvent(long time,List<OutgoingMessage> messages)
    {
        super(time);
        mMessages=messages;
    }

    private MIDIEvent(long time,OutgoingMessage message)
    {
        super(time);
        mMessages=new ArrayList<>();
        mMessages.add(message);
    }

    public MIDIEvent(long time, List<OutgoingMessage> messages, EventOffset offset)
    {
        this(time,messages);
        mOffset=offset;
    }

    public MIDIEvent(long time, OutgoingMessage message, EventOffset offset)
    {
        this(time,message);
        mOffset=offset;
    }
}

File: app/src/main/java/com/stevenfrew/beatprompter/event/MIDIEvent.kt
package com.stevenfrew.beatprompter.event

import com.stevenfrew.beatprompter.midi.EventOffset
import com.stevenfrew.beatprompter.midi.OutgoingMessage

class MIDIEvent : BaseEvent {
    @JvmField var mMessages: List<OutgoingMessage>
    @JvmField var mOffset: EventOffset?=null

    constructor(time: Long, messages: List<OutgoingMessage>) : super(time) {
        mMessages = messages
    }

    private constructor(time: Long, message: OutgoingMessage) : super(time) {
        mMessages = listOf(message)
    }

    constructor(time: Long, messages: List<OutgoingMessage>, offset: EventOffset) : this(time, messages) {
        mOffset = offset
    }

    constructor(time: Long, message: OutgoingMessage, offset: EventOffset) : this(time, message) {
        mOffset = offset
    }
}